zebra/zebra_nhg.c

```diff
@@ -2661,39 +2664,27 @@ void zebra_nhg_dplane_result(struct zebra_dplane_ctx *ctx)
 			SET_FLAG(nhe->flags, NEXTHOP_GROUP_VALID);
 			SET_FLAG(nhe->flags, NEXTHOP_GROUP_INSTALLED);
 			zebra_nhg_handle_install(nhe);
-		} else
+
+			/* If daemon nhg, send it an update */
+			if (nhe->id >= ZEBRA_NHG_PROTO_LOWER)
+				zsend_nhg_notify(nhe->type, nhe->zapi_instance,
+						 nhe->zapi_session, nhe->id,
+						 ZAPI_NHG_INSTALLED);
+		} else {
+			/* If daemon nhg, send it an update */
+			if (nhe->id >= ZEBRA_NHG_PROTO_LOWER)
+				zsend_nhg_notify(nhe->type, nhe->zapi_instance,
+						 nhe->zapi_session, nhe->id,
+						 ZAPI_NHG_FAIL_INSTALL);
+
 			flog_err(
 				EC_ZEBRA_DP_INSTALL_FAIL,
 				"Failed to install Nexthop ID (%u) into the kernel",
 				nhe->id);
+		}
 		break;
-	case DPLANE_OP_ROUTE_INSTALL:
-	case DPLANE_OP_ROUTE_UPDATE:
-	case DPLANE_OP_ROUTE_DELETE:
-	case DPLANE_OP_ROUTE_NOTIFY:
-	case DPLANE_OP_LSP_INSTALL:
-	case DPLANE_OP_LSP_UPDATE:
-	case DPLANE_OP_LSP_DELETE:
-	case DPLANE_OP_LSP_NOTIFY:
-	case DPLANE_OP_PW_INSTALL:
-	case DPLANE_OP_PW_UNINSTALL:
-	case DPLANE_OP_SYS_ROUTE_ADD:
-	case DPLANE_OP_SYS_ROUTE_DELETE:
-	case DPLANE_OP_ADDR_INSTALL:
-	case DPLANE_OP_ADDR_UNINSTALL:
-	case DPLANE_OP_MAC_INSTALL:
-	case DPLANE_OP_MAC_DELETE:
-	case DPLANE_OP_NEIGH_INSTALL:
-	case DPLANE_OP_NEIGH_UPDATE:
-	case DPLANE_OP_NEIGH_DELETE:
-	case DPLANE_OP_VTEP_ADD:
-	case DPLANE_OP_VTEP_DELETE:
-	case DPLANE_OP_RULE_ADD:
-	case DPLANE_OP_RULE_DELETE:
-	case DPLANE_OP_RULE_UPDATE:
-	case DPLANE_OP_NEIGH_DISCOVER:
-	case DPLANE_OP_BR_PORT_UPDATE:
-	case DPLANE_OP_NONE:
+
+	default:
```

Comment:
I hate defaults.. boo.  People will never know to update this if there is a dfeault

Reply:
heh - I knew you'd spot this! but really: this is only called from another switch statement that _does_ have every value in it. by the time the code gets here, we're only processing a couple of values, and it just looked so awkward to be forcing folks to maintain the complete set of values here. I get that you want to be careful not to lose a case in switch clauses where it matters, but I don't think this is one of those places?

Reply:
yep because I'm going to have to have the argument about it when someone else adds a new switch statement that just cuts-n-pastes from this.  Additionally the point of the non-use of the default is that it *automatically* tells you where you must touch code.  so this is a no go from me

Reply:
ok, pushed the values back in...